detox/src/client/Client.js

```diff
@@ -73,16 +73,23 @@ class Client {
 
     try {
       if (this.isConnected) {
-        await this.sendAction(new actions.Cleanup(this._successfulTestRun));
+        this._scheduleAppTermination();
+
+        try {
+          await this.sendAction(new actions.Cleanup(this._successfulTestRun));
+        } catch (e) {
+          log.error({ event: 'ERROR' }, DetoxRuntimeError.format(e));
```

Comment:
TODO: add some extra description of the failure, e.g.:

```js
          log.error({ event: 'ERROR' }, DetoxRuntimeError.format(new DetoxRuntimeError({
            message: 'Failed to conduct the app cleanup over web sockets due to the error.',
            debugInfo: e,
          })));
```

Reply:
I decided to enhance `AsyncWebSocket` so that it has a `timeout=N` option you can attach to the request.